operator/v1/tests/consoles.operator.openshift.io/AAA_ungated.yaml

```diff
@@ -296,3 +296,179 @@ tests:
+      expectedError: 'spec.customization: Invalid value: "object": Only one of logos or customLogoFile can be set'
+    - name: Should not allow invalid file extension in source.configMap.key
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: Console
+        spec:
+          customization:
+            logos:
+              - type: Masthead
+                themes:
+                  - mode: Dark
+                    source:
+                      from: ConfigMap
+                      configMap:
+                        name: custom-logos-config
+                        key: invalid-key.invalidextension
+      expectedError: 'spec.customization.logos[0].themes[0].source.configMap.key: Invalid value: "string": The ConfigMap key must end with a valid file extension (2 to 5 letters).'
+    - name: Should not allow invalid characters in source.configMap.name
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: Console
+        spec:
+          customization:
+            logos:
+              - type: Masthead
+                themes:
+                  - mode: Dark
+                    source:
+                      from: ConfigMap
+                      configMap:
+                        name: invalid name!
+                        key: masthead-dark-logo.svg
+      expectedError: "spec.customization.logos[0].themes[0].source.configMap.name: Invalid value: \"string\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+    - name: Should fail validation if ConfigMap is not set when From is ConfigMap
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: Console
+        spec:
+          customization:
+            logos:
+              - type: Masthead
+                themes:
+                  - mode: Dark
+                    source:
+                      from: ConfigMap
+      expectedError: "spec.customization.logos[0].themes[0].source: Invalid value: \"object\": configMap is required when from is 'ConfigMap', and forbidden otherwise."
+    - name: Should fail validation if ConfigMap is set and From is invalid source
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: Console
+        spec:
+          customization:
+            logos:
+              - type: Masthead
+                themes:
+                  - mode: Dark
+                    source:
+                      from: InvalidSource
+                      configMap:
+                        name: custom-logos-config
+                        key: masthead-dark-logo.svg
+      expectedError: "[spec.customization.logos[0].themes[0].source.from: Unsupported value: \"InvalidSource\": supported values: \"ConfigMap\", <nil>: Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]"
+      
```

Comment:
The final line has whitespace on it, please remove and get rid of the red mark at the end of the file. The final character should be `\n`

Reply:
This is still an issue

Reply:
Hope this look right now?